README.md

```diff
@@ -29,3 +29,60 @@ The color coding needs to be printed as a reference manual for wiring personnel.
 This manual is a mapping from the color-names to the corresponding numbers.
 Add a function that would format the color coding in a form that someone can print.
 Also resolve any coding guideline violations.
+
+## My Assignment Details:(Work in progress)
+--Excuse me for my grammatical mistakes.
+Wanted to write this file to let my intention behind splitting into a readable format.
+
+Divided it into the following units
+1. MajorColor
+2. MinorColor
+3. ColorPair
+4. ColorPairEvaluate
+5. Telecomm_ColorCode
+
+### MajorColor
+This unit contains all the handling of MajorColor.
+String array of MajorColor is moved into this unit
+and an interface is provided for other units. So its
+the unit responsibility to properly order the strings
+and enum in the same fashion.
+Added a macro which tells maximum number of characters
+in MajorColor. Updating this macro falls under this 
+units responsibility.
+
+### MinorColor
+This unit contains all the handling of Minor Color.
+The reasoning is same as MajorColor Unit.
+
+### ColorPair
+This unit contains all the handling at color pair level.
+Maximum Characters in a color pair Macro is now calculated
+with a macro from MajorColorunit , MinorColor unit and
+extra buffer length.
+So if there is any change in the length of major color and 
+minor color length. no need to change other modules.
+Didnt wanted to keep size as constant instead a configurable macro
+Wanted to place it in this unit as it is an colorpair level length
+
+### ColorPairEvaluate
+This was some confusing unit for me. As this only provides APIs for
+evaluation.
+Even though this is at colorpair level i wanted to have seperate unit
+for evaluation API. So if any more evaluation APIs are to come they
+will sit here.
+Renamed the APIs from test...() to Evaluate...() as it sounded nice.
+
+### Telecomm_ColorCode
+I did not find a good name for this unit as the purpose of this file
+was a little ambigious. Contains code that evaluates the other code.
+Contains a function that prints the manual.
```

Comment:
`StartEvaluation.c` instead of Telecom_ColorCode? It starts the evaluation.
But the `PrintColorCodeManual` would be a misfit there. It would help if each file has a single level of abstraction

Reply:
But the problem is int main() needs to call this printcolorcodemanual() and it also calls up for evaluations.... 
I am not sure if i am complicating things( idea from my dailylife ).. but then we have a task scheduler.c  
and we have different different functions like 
```   
   EvaluateColorPairToNumber(MajorColor_BLACK, MinorColor_ORANGE, 12);  
   EvaluateColorPairToNumber(MajorColor_VIOLET, MinorColor_SLATE, 25);  
   PrintColorCodeManual();  
```
 and taskschedular.c can contain main() function in which they all can be a part of.
i would not name it StartEvaluation.c though as start is misleading. i think Evaluate.c should be sufficient.
But then again if Evaluate.c is going to be present thne i would not like ot have colorpairEvaluate,c  then we might end up having two files with more or less same intention.

So this is some confusion which i didnot knew on how to keep in readme.md